### PR TITLE
GLTFLoader: Top-down loading for cameras.

### PR DESCRIPTION
**Pull request by donmccurdy**
Part 2 for #11924.

/cc @takahirox

### CODE REVIEW

**Comment by takahirox** [comment]
Sorry for the late reviewing.
Would you please fix the conflict?
I'll review once you've done.

**Comment by donmccurdy** [comment]
No worries, thanks! it is rebased now.

File: examples/js/loaders/GLTFLoader.js [hunk 1]
```diff
@@ -2028,43 +2023,39 @@ THREE.GLTFLoader = ( function () {
...
-		var json = this.json;
+	GLTFParser.prototype.loadCamera = function ( cameraIndex ) {
 
-		return _each( json.cameras, function ( camera ) {
+		var camera;
+		var cameraDef = this.json.cameras[ cameraIndex ];
+		var params = cameraDef[ cameraDef.type ];
 
-			var _camera;
+		if ( !params ) {
```
Comment (takahirox) [inline examples/js/loaders/GLTFLoader.js]: Very minor thing tho, you need an space after `!` as the coding style.

`if ( ! params ) {`

File: examples/js/loaders/GLTFLoader.js [hunk 2]
```diff
@@ -1341,43 +1342,19 @@ THREE.GLTFLoader = ( function () {
...
-			}
-
-			var scene = json.scene !== undefined ? dependencies.scenes[ json.scene ] : scenes[ 0 ];
-
-			var cameras = [];
-
-			dependencies.cameras = dependencies.cameras || [];
+			var scenes = dependencies.scenes;
+			var scene = scenes[ json.scene || 0 ];
+			var animations = dependencies.animations;
```
Comment (takahirox) [inline examples/js/loaders/GLTFLoader.js]: Is this intended change that `animations` will be `undefined` if glTF doesn't have animations while it was `[]`?

**Comment by takahirox** [comment]
Looks good other than two minor things as I commented.

**Review by takahirox — APPROVED** [review]

**Comment by donmccurdy** [comment]
Done, thanks!

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**